src/term/check/repeated_bind.rs

```diff
@@ -0,0 +1,70 @@
+use crate::term::{Ctx, Name, Term};
+use std::{collections::HashSet, fmt::Display};
+
+#[derive(Debug, Clone)]
+pub enum RepeatedBindWarn {
+  Rule(Name),
+  Match(Name),
+}
+
+impl Display for RepeatedBindWarn {
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    match self {
+      RepeatedBindWarn::Rule(bind) => write!(f, "Repeated bind inside rule pattern: '{bind}'."),
+      RepeatedBindWarn::Match(bind) => write!(f, "Repeated bind inside match arm: '{bind}'."),
```

Comment:
Actually, can we add an explanation of what this means here? Something like "The last variable will shadow all the others" or "Only the last bind will be available"

Reply:
I don't know why, but actually its only the first one that is binded

Reply:
ex:
```
Foo a a a = a
main = (Foo 1 2 3)
```
returns `1`

Reply:
Maybe we can leave it like this here and fix and improve the message in another PR, when it's correct.

Reply:
> Yeah, I'm pretty sure it's a bug, since `(Foo a b c) = ...` should desugar to `Foo = @a @b @c ...` and the same for more complex matches

but it does desugar to that, the problem is only that the var binds to the first:
```
(Foo) = λa λb λc a
```

Reply:
But if we have `(Foo a a a) = a` it'll desugar to `Foo = @a @a @a a`, the last lambda sugars the previous ones

Reply:
Ah, I see what you mean

Reply:
The problem has to do with the matches, since it actually desugars to `(Foo) = λ%x0 λ%x1 λ%x2 match %x0, %x1, %x2 { a a a: a }`